### PR TITLE
RF: Remove much of the hardcoded `T1w` fields in favor of `anat`

### PR DESCRIPTION
**Pull request by mgxd**
This is a supplemental PR to https://github.com/nipreps/nibabies/pull/360 - since nibabies also relies on smriprep, but may be processing in T2w space, this PR generalize many of the workflows/nodes/fields to account for other anatomical references.

### CODE REVIEW

**Comment by mgxd** [comment]
@effigies This should be ready for review. I've tried to limit any breaking changes, but regardless this will need to be included in a new minor release

File: smriprep/workflows/anatomical.py [hunk 1]
```diff
@@ -1375,7 +1365,7 @@ def init_anat_template_wf(
     longitudinal: bool,
     omp_nthreads: int,
     num_files: int,
-    contrast: str,
+    image_type: ty.Literal['T1w', 'T2w'] = 'T1w',
```
Comment (effigies) [inline smriprep/workflows/anatomical.py:1368]: Does this need a default for some reason?

File: smriprep/workflows/outputs.py [hunk 2]
```diff
@@ -538,44 +571,44 @@ def init_ds_template_registration_wf(
...
 
-    ds_t1w2std_xfm = pe.MapNode(
+    ds_anat2std_xfm = pe.MapNode(
         DerivativesDataSink(
             base_directory=output_dir,
             mode='image',
             suffix='xfm',
             dismiss_entities=['desc'],
-            **{'from': 'T1w'},
+            **{'from': image_type},
```
Comment (effigies) [inline smriprep/workflows/outputs.py:593]: We should think about whether the subject-native space should continue to be called `T1w`. We could call it `anat`, so we'd have `from-anat_to-MNI_xfm.h5` and `space-anat_desc-preproc_bold.nii.gz`.

But that does not need to be decided now, and it's better not to change the output names until we're sure that's what we want.

File: smriprep/workflows/outputs.py [hunk 3]
```diff
@@ -731,6 +765,8 @@ def init_ds_surfaces_wf(
                 ds_surf.inputs.desc = 'reg'
                 if surf == 'sphere_reg_fsLR':
                     ds_surf.inputs.space = 'fsLR'
+                elif surf == 'sphere_reg_dhcpAsym':
+                    ds_surf.inputs.space = 'dhcpAsym'
```
Comment (effigies) [inline smriprep/workflows/outputs.py:769]: Are you using msmsulc to align to dhcpAsym, or is that what MCRIBs registers to by default?

**Review by effigies — APPROVED** [review]
LGTM. Left some non-blocking comments.

File: smriprep/workflows/anatomical.py (hunk 1 above)
Comment (mgxd) [reply smriprep/workflows/anatomical.py:1368]: I was intending to minimize changes in fmriprep

File: smriprep/workflows/outputs.py (hunk 3 above)
Comment (mgxd) [reply smriprep/workflows/outputs.py:769]: MCRIBs already produces the registration to dhcpAsym, which makes it easier to get to CIFTI space.

We aren't running msmsulc yet, mostly due to my lack of clarity of the differences between dhcpAsym / fsLR surfaces.

File: smriprep/workflows/anatomical.py (hunk 1 above)
Comment (effigies) [reply smriprep/workflows/anatomical.py:1368]: I'd rather rip off the bandaid. Anything that you think should be generic, let's make it generic and deal with it now.

File: smriprep/workflows/outputs.py (hunk 2 above)
Comment (mgxd) [reply smriprep/workflows/outputs.py:593]: I like calling it `anat`, but I feel it still carries some ambiguity. Perhaps having some metadata (either in the sidecar or a separate, generated file) covering exactly this space is (image dimensions, files used to derive it, etc) may be helpful.

File: smriprep/workflows/anatomical.py (hunk 1 above)
Comment (mgxd) [reply smriprep/workflows/anatomical.py]: all set - probably can squash all of this when merging

**Pull request merged**